nrich-encrypt/src/main/java/net/croz/nrich/encrypt/service/DefaultDataEncryptService.java

```diff
@@ -101,24 +106,41 @@ private void encryptDecryptNestedValue(EncryptionContext encryptionContext, Obje
 
     private void encryptDecryptValue(EncryptionContext encryptionContext, Object objectContainingFieldsToEncryptOrDecrypt, String propertyName, EncryptionOperation operation) {
         Object value = getPropertyValueByPath(objectContainingFieldsToEncryptOrDecrypt, propertyName);
+        Object result = encryptDecryptValue(encryptionContext, value, operation);
+
+        if (result == null) {
+            return;
+        }
 
+        setPropertyValueByPath(objectContainingFieldsToEncryptOrDecrypt, propertyName, result);
+    }
+
+    private Object encryptDecryptValue(EncryptionContext encryptionContext, Object value, EncryptionOperation operation) {
         if (value instanceof String) {
             String textToEncryptOrDecrypt = (String) value;
-            String encryptedValue = encryptDecryptText(encryptionContext, textToEncryptOrDecrypt, operation);
-
-            setPropertyValueByPath(objectContainingFieldsToEncryptOrDecrypt, propertyName, encryptedValue);
 
+            return encryptDecryptText(encryptionContext, textToEncryptOrDecrypt, operation);
         }
-        else if (value instanceof Collection && ((Collection<?>) value).stream().allMatch(String.class::isInstance)) {
+        else if (isSupportedCollection(value) && ((Collection<?>) value).stream().allMatch(String.class::isInstance)) {
             @SuppressWarnings("unchecked")
             Collection<String> textToEncryptOrDecryptList = (Collection<String>) value;
+            Collector<? super String, ?, ?> collector = value instanceof Set ? Collectors.toSet() : Collectors.toList();
 
-            Collection<String> encryptedValueList = textToEncryptOrDecryptList.stream()
+            return textToEncryptOrDecryptList.stream()
                 .map(textToEncryptOrDecrypt -> encryptDecryptText(encryptionContext, textToEncryptOrDecrypt, operation))
-                .collect(Collectors.toList());
+                .collect(collector);
+        }
+        else if (value instanceof String[]) {
+            String[] textToEncryptOrDecryptList = (String[]) value;
 
-            setPropertyValueByPath(objectContainingFieldsToEncryptOrDecrypt, propertyName, encryptedValueList);
+            return Arrays.stream(textToEncryptOrDecryptList)
+                .map(textToEncryptOrDecrypt -> encryptDecryptText(encryptionContext, textToEncryptOrDecrypt, operation))
+                .toArray(String[]::new);
         }
+
+        log.warn("Unable to encrypt/decrypt property, check if the specified path is correct and if the specified property is a string");
```

Comment:
Can we enhance the warning message to be more precise (two cases: just encrypt or just decrypt)? As we already have the `EncryptionOperation` here.